internal/otc_wrapper.go

```diff
@@ -168,6 +170,29 @@ func (c *OtcWrapper) GetElbIdNameMapping() (map[string]string, error) {
 	return result, nil
 }
 
+// Add in the DDS Service client here (openstack client.go line 719) - David
+func (c *OtcWrapper) GetDdsIdNameMapping() (map[string]string, error) {
+	opts := golangsdk.EndpointOpts{Region: "eu-de"}
+	ddsClient, err := openstack.NewDDSServiceV3(c.providerClient, opts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	ddsListResponse, err := ddsInstances.List(ddsClient, ddsInstances.ListInstanceOpts{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := map[string]string{}
+	for _, instance := range ddsListResponse.Instances {
+		result[instance.Groups[0].Nodes[0].Id] = instance.Name
```

Comment:
In this part you would need to iterate over `instance.Groups` and `instance.Groups[i].Nodes`. I tested the code with some DDS instances. 

- With a single-node DDS it works perfectly fine. 
- With a replica-set DDS this does not work because the nodes array contains then multiple elements. See the first screenshot, please.
- With a cluster DDS this also does not work because then the Groups array contains multiple elements. See the second screenshot, please.

<img width="426" alt="image" src="https://github.com/iits-consulting/otc-prometheus-exporter/assets/10364991/5de17f03-8d26-4fa1-aa05-567e8f8667c6">

<img width="426" alt="image" src="https://github.com/iits-consulting/otc-prometheus-exporter/assets/10364991/aa98a67f-a4de-4ba5-aee9-e0fedc6d8cbd">